frontend/src/reusable/BasicForm.tsx

```diff
@@ -0,0 +1,98 @@
+import { useForm } from "@mantine/form";
+import { Button, Fieldset, Group } from "@mantine/core";
+import { PromptDefinition } from "./PromptTypes";
+import { ReactNode } from "react";
+
+type PromptCollectorProps = { promptDefinitions: PromptDefinition[] };
+
+function updateImageBlur(
+    imageName: string,
+    blurQuantity: number = 9,
+    blurScaling: number = 7,
+) {
+    const image = document.getElementById(imageName);
+    if (!image) {
+        console.error(`Cannot find image with name ${imageName}`);
+    } else {
+        image?.classList.remove("blur-3xl");
+        image.style.filter = `blur(${blurQuantity * blurScaling}px)`;
+    }
+}
+
+type FormValues = {
+    prompts: {
+        [key: string]: {
+            data: string;
+        };
+    };
+};
+
+export function BasicForm(props: PromptCollectorProps) {
+    let formValues: Map<string, any> = new Map<string, any>();
+
+    let children: ReactNode[] = [];
+
+    const form = useForm({
+        mode: "uncontrolled",
+        initialValues: {} as FormValues,
+        onValuesChange(values, previous) {
+            if (
+                Object.keys(previous).length != 0 &&
+                Object.keys(values).length != 0
+            ) {
+                const niceValues = new Map(Object.entries(values.prompts));
+                const changedValues = [...niceValues.values()].filter(
+                    (value) => value.data != "",
+                );
```

Comment:
why the map conversion? to get rid of duplicate keys? all the keys should be unique i think

Reply:
much better, thank you